[Keycloak CI] - User Federation Tests - LDAPUserProfileTest.testMultipleLDAPProviders

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

```
2025-01-09 11:55:55  -- [ERROR] org.keycloak.testsuite.federation.ldap.LDAPUserProfileTest.testMultipleLDAPProviders  Time elapsed: 5.992 s  <<< ERROR!
2025-01-09 11:55:55  -- jakarta.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.handleErrorStatus(ClientInvocation.java:250)
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.extractResult(ClientInvocation.java:216)
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.proxy.extractors.BodyEntityExtractor.extractEntity(BodyEntityExtractor.java:59)
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:136)
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:103)
2025-01-09 11:55:55  -- 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:102)
2025-01-09 11:55:55  -- 	at jdk.proxy2/jdk.proxy2.$Proxy104.runOnServer(Unknown Source)
2025-01-09 11:55:55  -- 	at org.keycloak.testsuite.client.KeycloakTestingClient$Server.run(KeycloakTestingClient.java:165)
2025-01-09 11:55:55  -- 	at org.keycloak.testsuite.federation.ldap.LDAPUserProfileTest.testMultipleLDAPProviders(LDAPUserProfileTest.java:261)
```

https://keycloak-jenkins/view/Keycloak/job/keycloak-continuous-testing/59/execution/node/591/log/

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPUserProfileTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPUserProfileTest.java
@@ -270,9 +270,13 @@ public class LDAPUserProfileTest extends AbstractLDAPTest {
             testRealm.addComponentModel(ldapModel);
             LDAPStorageProvider ldapProvider = LDAPTestUtils.getLdapProvider(session, ldapModel);
 
-            // if AD, create new OU in a base DN because users.ldif is ignored for AD
-            if (LDAPConstants.VENDOR_ACTIVE_DIRECTORY.equals(ldapModel.getConfig().getFirst(LDAPConstants.VENDOR))) {
+            // if AD or RHDS, create new OU in a base DN because users.ldif is ignored for AD/RHDS
+            String vendor = ldapModel.getConfig().getFirst(LDAPConstants.VENDOR);
+            if (LDAPConstants.VENDOR_ACTIVE_DIRECTORY.equals(vendor)) {
                 LDAPTestUtils.addLdapOUinBaseDn(ldapProvider, "OtherPeople2");
+                LDAPTestUtils.removeAllLDAPUsers(ldapProvider, testRealm);
+            } else if (LDAPConstants.VENDOR_RHDS.equals(vendor)) {
+                LDAPTestUtils.addLdapOUinBaseDn(ldapProvider, "OtherPeople");
                 LDAPTestUtils.removeAllLDAPUsers(ldapProvider, testRealm);
             }
             LDAPObject john = LDAPTestUtils.addLDAPUser(ldapProvider, testRealm, "anotherjohn", "AnotherJohn", "AnotherDoe", "anotherjohn@email.org", null, "1234");

